src/libstd/collections/hash/table.rs

```diff
@@ -897,8 +898,9 @@ unsafe impl<K: Send, V: Send> Send for IntoIter<K, V> {}
 
 /// Iterator over the entries in a table, clearing the table.
 pub struct Drain<'a, K: 'a, V: 'a> {
-    table: &'a mut RawTable<K, V>,
+    table: Shared<RawTable<K, V>>,
```

Comment:
`&mut T` is invariant in `T`. But we're reading `(K, V)` out of the table, not writing to it, so covariance should be safe. This isn't conveyed by `&mut RawTable<K, V>` alone, because the compiler has to assume that such a reference could be written to.